src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1481,7 +1481,7 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
                 expression = null;
                 if (node is IdentifierNameSyntax identifier)
                 {
-                    var type = BindNativeIntegerSymbolIfAny(identifier, diagnostics);
+                    var type = IsInsideNameof ? null : BindNativeIntegerSymbolIfAny(identifier, diagnostics);
```

Comment:
Why things just doesn't flow naturally for native ints just like dynamic?
Is there a deeper problem that I'm just work-arounding it?

Reply:
This code path is under the umbrella of `lookupResult.Kind == LookupResultKind.Empty`, which as far as I understand happens when there is no symbols declared with that identifier name.

If there is no symbols declared with the name "nint" or "nuint", what does this line is trying to do?

Reply:
This code path is under the umbrella of `lookupResult.Kind == LookupResultKind.Empty`, which as far as I understand happens when there is no symbols declared with that identifier name.

If there is no symbols declared with the name "nint" or "nuint", why/when the binding is needed?

Reply:
You could debug nameof(dynamic) to see what cause it to be an error

Reply:
@alrz I did.
The reason is simply because `expression` remains null as it was declared in line 1481.
It's not the case with native int because after the binding, expression isn't null anymore.

Is `dynamic` not requiring any binding, that's why it's different?